src/pyclass/create_type_object.rs

```diff
@@ -386,6 +387,8 @@ impl PyTypeBuilder {
         // Safety: We've correctly setup the PyType_Spec at this point
         let type_object: Py<PyType> =
             unsafe { Py::from_owned_ptr_or_err(py, ffi::PyType_FromSpec(&mut spec))? };
+        // Safety: `name` is a valid pointer obtained from `CString::into_raw` in `py_class_qualified_name`
+        drop(unsafe { CString::from_raw(name) });
```

Comment:
Rather than use unsafety here let's change `py_class_qualified_name` to return a `CString`. That way we can also flip the check around and instead `mem::forget(name)` on Python < 3.11.